mds/authent/public_api.py

```diff
@@ -53,22 +71,43 @@ def get_revocation_list() -> List[str]:
     return token_ids
 
 
-def create_application(name, owner=None, user=None, grant=None, scopes=None):
+def create_application(name, owner=None, grant=None, scopes=None):
     grant = grant or models.Application.GRANT_CLIENT_CREDENTIALS
-    return models.Application.objects.create(
+    app, created = models.Application.objects.get_or_create(
         name=name,
-        redirect_uris="http://example.com",
         client_type=models.Application.CLIENT_CONFIDENTIAL,
         authorization_grant_type=grant,
         scopes=scopes,
-        user=user,
         owner=owner,
     )
+    return {
+        "name": name,
+        "client_id": app.client_id,
+        "client_secret": app.secret,
+        "client_type": models.Application.CLIENT_CONFIDENTIAL,
+        "grant_type": grant,
+        "scopes": scopes,
+        "created": created,
+    }
 
 
 def revoke_application(owner_id):
-    models.Application.objects.filter(owner=owner_id).update(scopes=[])
+    app_qs = models.Application.objects.filter(owner=owner_id)
+    updated = app_qs.update(scopes=[])
+    if not updated:
+        raise NoApplicationForOwner()
+    _revoke_tokens(
+        models.AccessToken.objects.filter(
+            application_id__in=app_qs.values_list("id", flat=True)
+        )
+    )
 
 
 def delete_application(owner_id):
-    models.Application.objects.filter(owner=owner_id).delete()
```

Comment:
@alexandrelm : Not sure about this one. What do you do with existing tokens if any ? Delete them ? But since they are JWT you may want to keep them.